### ISSUE TITLE
Deployment Resource ID missing when failure is due to template rather resource

### ISSUE THREAD

**Issue by slavizh** [Azure/deployment-stacks#108]
When deployment fails due to resource not being able to be provisioned we get nice error in FailedResources property and Error property but when there is a failure for template we do not have deployment resource ID, we only get the error from that deployment but not in which deployment exactly occurred (see screenshot). Knowing which deployment is can help you save time to understand where the problem is without having to browse the portal.

![image](https://github.com/Azure/deployment-stacks/assets/11890856/349f5f0b-15bd-4597-9828-499143669d5e)

If we trace on portal we will see this and we can see which deployment the error is on:

![image](https://github.com/Azure/deployment-stacks/assets/11890856/b7b2614c-dc3a-43b6-8846-7ad311a9f62a)

**Comment by stuartko** [comment]
@slavizh Was this a nested deployment within your stack Deployment? Just to confirm, you are seeing $yourStack.deploymentId as empty?

**Comment by slavizh** [comment]
Yes, this is nested deployment. $yourStack.deploymentId  is not empty but when you have nested deployments the actual error is within the lowest deployment in the three. As you can see from the screenshot Target is missing there. I think that should be the resource ID of the deployment from where the error comes.

**Comment by stuartko** [comment]
@slavizh : Got it. Thanks for confirming. The "target" property is on our roadmap for errors, we're just awaiting a coordination between a deployments version API bump and an SDK release.

**Comment by torreymicrosoft** [comment]
Thanks for your patience — picked up as part of an open-issue review for the ARM Deployments team.

This is a reasonable diagnosability ask: when a stack's underlying deployment fails, the error should include the deployment resource ID (or a direct link) so you can jump straight to the failing deployment's operations instead of hunting for it. It's in the same error-surfacing area as #212, and we'll track it alongside that work. Closing as duplicate and tagging as feature ask.
